File: Assignment4/Q1.c
/////////////////////////////////////////////////////////////////////////////////
//
//      Include required header files
//
/////////////////////////////////////////////////////////////////////////////////
#include<stdio.h>

/////////////////////////////////////////////////////////////////////////////////
//
//      Function name:  MultFact
//      Input:          integer
//      Output:         integer
//      Description:    Calculates multiplication of factors of given number
//      Date:           24/05/2026
//      Author:         Pratik Subhash Mahale
//
/////////////////////////////////////////////////////////////////////////////////
int MultFact(
                int iNo             // Input number
            )
{
    int i = 0;                      // Counter variable
    int iMult = 1;                  // Variable to store multiplication

    // Convert negative number into positive
    if(iNo < 0)
    {
        iNo = -iNo;
    }

    // Loop to find factors
    for(i = 1; i <= iNo / 2; i++)
    {
        // Check whether i is factor
        if(iNo % i == 0)
        {
            iMult = iMult * i;      // Multiply factors
        }
    }

    return iMult;                   // Return multiplication
}

/////////////////////////////////////////////////////////////////////////////////
//
//      Application to calculate multiplication of factors
//
/////////////////////////////////////////////////////////////////////////////////
int main()
{
    int iValue = 0;                 // Variable to store input number
    int iRet = 0;                   // Variable to store return value

    printf("Enter number\n");

    scanf("%d", &iValue);

    // Function call
    iRet = MultFact(iValue);

    // Display result
    printf("%d", iRet);

    return 0;                       // Successful termination
}

/////////////////////////////////////////////////////////////////////////////////
//
//      Input:      12
//      Output:     144
//
/////////////////////////////////////////////////////////////////////////////////

File: Assignment21/Q1.java
///////////////////////////////////////////////////////////////////
//
//      File Name : Q1.java
//      Project   : Logic Building Assignment 21
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Class Name : ProductDigitLogic
//      Method Name: productOfDigits
//      Input      : Integer
//      Output     : Void
//      Description: Calculate the product of
//                   all digits of a number
//      Date       : 08/06/2026
//      Author     : Pratik Subhash Mahale
//
///////////////////////////////////////////////////////////////////

class ProductDigitLogic
{
    void productOfDigits(int num)
    {
        int digit = 0;          // Store extracted digit
        int iProduct = 1;       // Store multiplication result

        // Convert negative number to positive
        if(num < 0)
        {
            num = -num;
        }

        // Calculate product of digits
        while(num != 0)
        {
            digit = num % 10;

            iProduct = iProduct * digit;

            num = num / 10;
        }

        // Display result
        System.out.println("Product of digits is : " + iProduct);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Main Class
//
///////////////////////////////////////////////////////////////////

public class Q1
{
    public static void main(String args[])
    {
        // Create object of ProductDigitLogic class
        ProductDigitLogic obj = new ProductDigitLogic();

        // Function call
        obj.productOfDigits(234);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Input  : 234
//
//      Output :
//      Product of digits is : 24
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Input  : 105
//
//      Output :
//      Product of digits is : 0
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Time Complexity : O(d)
//
//      Space Complexity : O(1)
//
//      Where d = Number of digits
//
///////////////////////////////////////////////////////////////////

